server/lib/chat/chat-service.ts

```diff
@@ -767,12 +808,22 @@ export default class ChatService {
 
   private subscribeUserToChannel(
     userSockets: UserSocketsGroup,
-    channelInfo: ChannelInfo,
+    channelInfo: FullChannelInfo,
     channelPermissions: ChannelPermissions,
   ) {
     userSockets.subscribe<ChatInitEvent>(getChannelPath(channelInfo.id), () => ({
       action: 'init3',
-      channelInfo,
+      channelInfo: {
```

Comment:
This code is all broken, you need to be VERY careful about stuff like this. You have subscribed a user to the version of the channel that existed when they connected to the server. This object will NEVER be updated. 

You need to write an async method here that retrieves this on demand, passing in the existing object to the subscription does not work